File: app/src/main/java/com/hrithikvish/cbsm/ViewPagerPostsAndSavedAdapter.java
package com.hrithikvish.cbsm;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.fragment.app.FragmentStatePagerAdapter;

//tab layout adapter
public class ViewPagerPostsAndSavedAdapter extends FragmentStatePagerAdapter {
    public ViewPagerPostsAndSavedAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        if(position == 0) {
            return new ProfilePostsFragment();
        } else {
            return new ProfileSavedFragment();
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if(position == 0) {
            return "Posts";
        } else {
            return "Saved";
        }
    }

    @Override
    public int getCount() {
        return 2;
    }
}

File: app/src/main/java/com/hrithikvish/cbsm/adapter/ViewPagerPostsAndSavedAdapter.kt
package com.hrithikvish.cbsm.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.hrithikvish.cbsm.ProfilePostsFragment
import com.hrithikvish.cbsm.ProfileSavedFragment

//tab layout adapter
class ViewPagerPostsAndSavedAdapter(fm: FragmentManager) :
    FragmentStatePagerAdapter(fm) {
    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            ProfilePostsFragment()
        } else {
            ProfileSavedFragment()
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (position == 0) {
            "Your Items"
        } else {
            "Saved Items"
        }
    }

    override fun getCount(): Int {
        return 2
    }
}